### DISCUSSION TITLE
Jamulus for iOS not connecting through Tailscale

### DISCUSSION THREAD

**Discussion by a-singer** [jamulussoftware/jamulus#3311] [Bug Investigation]
Hi all,

I'm trying to use Jamulus via Tailscale on iOS. The program connects fine when over the local network, but does not connect at all through the tailscale network. When I enter the local IP on the local network, it connects at once. When I have it on another network, and enter the 100 IP tailscale gives me, it says "trying to connect" and never gets any further. Other programs connect through tailscale quite easily, all I have to do is enter the IP and they start talking to the server. Does the iOS version produce any logs I can look at, or can I somehow get more information on what's going wrong?

Thanks,

**Comment by dcorson-ticino-com** [comment]
I do not know what Tailscale is, but to my knowledge Jamulus does not work on iOS.

**Comment by rdica** [comment]
Hi @a-singer I searched on iOS App Store and the only Jamulus app I see is some third party app created by ICON DATA CENTRE LIMITED, it is not an official release by Jamulus. Is this what you're using, and if so, have you reached out to them for support?

**Reply by a-singer** [reply]
Hi,

Thanks very much for looking into this. I was not aware that the program I'm using, it is the one referred  to by @rdica wasn't official. I thought it was just a signed version of 

https://jamulus.io/wiki/Installation-for-iOS

so as to save myself signing the .ipa every week. I'm still not sure how to do this, but it may be worth doing. In any case, I will ask the provider for help with the third party program. Just to answer the question about Tailscale, it 

http://www.tailscale.com

is a sort of VPN replacement which is supposed to get around nat fairly easily. It does, in fact, but Jamulus, at least as it is produced by the third party on iOS, doesn't seem to work with it. Thanks to both of you for taking a look at this and the information.

**Reply by ann0see** [reply]
> so as to save myself signing the .ipa every week. I'm still not sure how to do this, but it may be worth doing. In any case, I will ask the provider for help with the third party program. Just to answer the question about Tailscale, it

Self signing is *currently* the only official way to use Jamulus on iOS. I have heard about tailscale before, never used it, but they may block (UDP) traffic?

>  Does the iOS version produce any logs I can look at, or can I somehow get more information on what's going wrong?

Yes. If you compile it yourself at least. Then it will show up in the macOS console/Xcode. I don't think that you'll see anything if it's a network related issue.

**Reply by ann0see** [reply]
> ICON DATA CENTRE LIMITED

This is indeed not official and they didn't respond as we contacted them. We [allow distribution on the AppStore](https://github.com/jamulussoftware/jamulus/blob/main/APPLEAPPSTORE.LICENCE.WAIVER) but as far as I know, they don't update.

I'd use the Koord client: https://koord.live/downloads since it's Jamulus compatible. The people behind this heavily customized app contribute and have had good track record.

**Reply by a-singer** [reply]
Thanks very much for taking the time to think with me about this, @ann0see So far as I know, tailscale doesn't block UDP at all, or at least isn't supposed to. If I may ask, what is the preferred method of signing the .ipa? All the instructions I'm finding are contradictory and old, and it would be nice to know how it can best be done now. The installation document doesn't specify any particular method. 

I'm very grateful for the recommendation, I had never heard of Koord before. I'm a bit confused, though, as to how to use it with my own jamulus server, they seem to want me to make a session on their own service. I should say that I'm blind and using voiceover, so it's quite possible that I'm missing how to do this and it's right there. They say, on their FAQ page, that you can create a session and join it with Jamulus, but I'm a bit lost as to how the two packages get the information they need to connect. Thanks again.

**Reply by ann0see** [reply]
Ok. For Koord, You can use the field where you'd input their server information to input your IP. Not sure how it's called in voice over, but if you tap join, the field shows up and says paste session link here... you would enter the IP or domain from Jamulus there.

**Reply by ann0see** [reply]
See https://gist.github.com/ann0see/8f354ed0ff154a2e9c47359eb8043020 for my personal sideloading docs. Highly unofficial of course

**Comment by ann0see** [comment]
To debug network issues, I'd check the App Store for traceroute apps. 

https://apps.apple.com/it/app/he-net-network-tools/id858241710
for example. Then check connectivity to your servers IP while being connected via tailscale. 

In any case, I'd recommend using port forwarding instead.

**Reply by a-singer** [reply]
I really can't thank you enough, @ann0see for both the advice and the link to your instructions on signing the file. It turns out that the defect was in the access tech, not in the app, either the app that Icon Center produced or the .ipa I just signed. The problems I'm having were caused by the fact that VoiceOver would sometimes say that the server address was there when it wasn't, sometimes say it wasn't there when it was, and sometimes not read it completely so that the name/IP was actually entered twice but I thought it was entered once. I just got lucky with the local address and unlucky with the tailscale address. When I got a sighted person to have a look, everything began to make sense. I have never seen this before so have put this up as a problem at discussion 3313. Thanks again for the links.

**Reply by ann0see** [reply]
> The problems I'm having were caused by the fact that VoiceOver would sometimes say that the server address was there when it wasn't, sometimes say it wasn't there when it was, and sometimes not read it completely so that the name/IP was actually entered twice but I thought it was entered once.

That's a bad accessibility issue IMO and needs investigation. Do you experience the same on macOS (if you own one)?

**Reply by ann0see** [reply]
You can also answer in https://github.com/orgs/jamulussoftware/discussions/3313

**Comment by QJKX** [comment]
Can you eliminate jamulus server problems by testing with an official (non-ios) client over Tailscale?

Is the jamulus server bound to all its IP addresses, including its Tailscale IP? It should listen to all addresses by default - specifying the lan address will mean it doesn't listen on it's wireguard ip. From memory with the server running,  `lsof -i|grep jamulus` will show you which ip addresses it's listening on.
